packages/discord-ticket-api/src/services/TicketService.ts

```diff
@@ -0,0 +1,43 @@
+import { Queue } from "bullmq";
+import { AppEnv } from "../env.js";
+
+export type OpenTicketRequest = {
+  title: string;
+  content: string;
+};
+
+export interface TicketService {
+  enqueue(request: OpenTicketRequest): Promise<{ jobId: string }>;
+}
+
+export type TicketJobData = {
+  channelId: string;
+  title: string;
+  content: string;
+};
+
+export class TicketQueueService implements TicketService {
+  private queue: Queue<TicketJobData, unknown, string>;
+  private env: AppEnv;
+
+  constructor(queue: Queue<TicketJobData, unknown, string>, env: AppEnv) {
+    this.queue = queue;
+    this.env = env;
+  }
+
+  async enqueue(request: OpenTicketRequest): Promise<{ jobId: string }> {
+    const job = await this.queue.add(
+      "open-ticket",
+      {
+        channelId: this.env.DISCORD_CHANNEL_ID,
+        title: request.title,
+        content: request.content,
+      },
+      {
+        removeOnComplete: { age: 3600, count: 1000 },
+        removeOnFail: { age: 24 * 3600, count: 1000 },
+      }
+    );
+    return { jobId: job.id as string };
+  }
```

Comment:
Jobs are enqueued with BullMQ defaults (attempts=1, no backoff). Any transient Discord/Redis error moves the job to failed (and then you delete it via removeOnFail), so tickets are silently lost unless manually retried. 

does setting attempts+backoff and retaining failures, e.g. attempts: 3, backoff: { type: "exponential", delay: 5000 }, removeOnFail: false (or longer retention), and consider alerting on failed jobs make sence?

Reply:
We never receive an acknowledgement from the tickettool system, and it's not possible to check this until we implement a custom solution to find tickets. That's why I just send it once optimistically, but I can maybe add retry logic.